src/consts/models.tsx

```diff
@@ -16,6 +17,12 @@ import {
   HDLegacyP2PKHWallet,
 } from '../../class';
 
+// don't change the order when addign the new version, the oldest user version is on the top, the newest on the bottom
+export enum USER_VERSIONS {
+  BEFORE_NOTIFICATIONS_WERE_ADDED = 'BEFORE_NOTIFICATIONS_WERE_ADDED',
+  AFTER_NOTIFICATIONS_WERE_ADDED = 'AFTER_NOTIFICATIONS_WERE_ADDED',
+}
```

Comment:
so what is your proposal about solving the problem?

Reply:
cannot we rely on the app version? Current version in stores is 6.0.1 so this one and every elder versions will be BEFORE_NOTIFICATIONS_WERE_ADDED

Reply:
but when we will release the app for every user the APP_VERSION will be equal to let's say 6.1.0, so how do you want to recognize if the user had 6.0.1 or earlier installed?

Reply:
I thought it's not about users versions but to control what to display on the "Add email address" screen - whether it's onboarding or settings flow. 
```
const getAddEmailInitialParams = () => {
     if (userVersion === USER_VERSIONS.BEFORE_NOTIFICATIONS_WERE_ADDED) {
       return getAppUpdateAddEmailParams();
     }
     return getOnboardingAddEmailParams();
   };
```
Maybe I misunderstood something.

Anyway, IMHO we should proceed as always, so move back the properties from `helpers/notification.ts` to components and pass there it directly to the route. I'm aware it's not perfect and our components become less and less readable but I also don't believe it's the right way to simplify and optimize our codebase. 

I think we should focus on increasing the readability of our components by focusing more on writing cleaner code. I know it's very general advice 😀

Reply:
USER_VERSION is related to what to display the user on AddEmailNotifcation (what texts, title, etc.), we have to somehow know if the user was in our app before are after the notification feature was added.
 To be honest, I don't think that it increases readability, I can always check what is inside the function, and the plus of extracting it to helpers is that we don't duplicate the code and we see that these part of the code are related.

Reply:
good point. Then keeping some flag in keychain like storedPin or transactionPassword seems to bee a cleaner solution but it is still just a way

Reply:
I was thinking about it but it also won't work as if the user exits the app as in the screenshot he should go back into this screen when opening the app again. So at this stage, there would be no way of distinguishing whether the user is updating the app or is the new user who should finish the onboarding process. 
Imho that's why Some sort of flag is required, and I thought it would be nice to this in a generic way, so it can be used also for airdrop, push notifications, etc.
![Zrzut ekranu 2021-01-25 o 10 57 10](https://user-images.githubusercontent.com/67273035/105690403-3d691b00-5efc-11eb-96e0-05ff3b9ae5e8.png)